### PR TITLE
For #1481. Use androidx runner in few more modules.

### PR DESCRIPTION
**Pull request by dector**
### Issue #1481 

  - Enable `includeAndroidResources` and `enableUnitTestBinaryResources` for `feature-sitepermissions` module.
  - Enable `includeAndroidResources` and `enableUnitTestBinaryResources` for `feature-readerview` module.
  - Enable `includeAndroidResources` and `enableUnitTestBinaryResources` for `service-experiments` module.
  - Turn on `enableUnitTestBinaryResources` globally (exclude few modules explicitly).
  - Use `AndroidJUnit4` as a test runner (from AndroidX Test Ext).

### Complexity

Easy (★☆☆)

### Pull Request checklist
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] ~~**Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one~~
- [x] ~~**Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features~~

### CODE REVIEW

**Comment by dector** [comment]
@Amejia481 I have one more commit for `service-experiments`. Should I create new PR or push it here?

**Comment by Amejia481** [comment]
@dector push it here!

**Comment by dector** [comment]
Oh, I've missed to unignore one test. One moment please. :man_facepalming:

**Comment by Amejia481** [comment]
No worries, just let me know when you finish :)

**Comment by dector** [comment]
@Amejia481 Done.

File: components/service/experiments/src/test/java/mozilla/components/service/experiments/ActiveExperimentTest.kt [hunk 1]
```diff
@@ -97,9 +99,27 @@ class ActiveExperimentTest {
         active = ActiveExperiment.load(context, experiments)
         assertEquals(experiments[1], active!!.experiment)
         assertEquals("branch3", active.branch)
+    }
+
+    @Test
+    fun `test load invalid experminet from storage`() {
```
Comment (travis79) [inline components/service/experiments/src/test/java/mozilla/components/service/experiments/ActiveExperimentTest.kt]: nit:
```suggestion
    fun `test load invalid experiment from storage`() {
```

Comment (dector) [reply components/service/experiments/src/test/java/mozilla/components/service/experiments/ActiveExperimentTest.kt]: Oops. I think uncle Sigmund can be proud of me. :sweat_smile: 

Thanks!

**Comment by dector** [comment]
@travis79 Are you working on `service-experiments`?

@Amejia481 Let's remove `service-experiments` from this PR? I suggest it should be merged after new implementation because applying this changes are much easier than rebasing new tests/fixes on 'em.

**Comment by travis79** [comment]
> @travis79 @Amejia481 Let's remove `service-experiments` from this PR? I suggest it should be merged after new implementation because applying this changes are much easier than rebasing new tests/fixes on 'em.

Thanks!  I expect the bulk of the changes I'm working on to be done in the next two weeks, I just know that I've touched the ExperimentsTest.kt quite a bit and it might be tricky.

**Comment by dector** [comment]
Probably it'll be good idea to include only minimal necessary changes without minor readability improvements.

**Comment by dector** [comment]
Done. With current changes rebasing on this revision should reduce conflicts in `service-experiments` almost to zero (some conflicts in imports are possible).

**Comment by travis79** [comment]
Thanks @dector, I hope to have the rest of the changes I'm working on done soon so I appreciate sparing me from the merge conflict.  😄

**Comment by dector** [comment]
@Amejia481 Pls pay attention to latest commit. I've turned on `enableUnitTestBinaryResources` globally. :)

**Pull request merged**